### ISSUE TITLE
Runing code stagnate on 7 scenes dataset

### ISSUE THREAD

**Issue by double344931987** [cvlab-dresden/DSAC#3]
Hi, 
I tried to run your code on the stair dataset using pre-train score regression CNN and end-to-end training using DSAC.
However, the performance does not match with the reported performance in your paper.

First, I have downloaded stairs scene and TSDF Volume from your website. 
Then, I run ../../core/build/train_obj -oscript train_obj.lua
 In train_obj.cpp , the trainingLimit is set to 300000, which is same as your report. However, it will cost lots of time to  finish all of the training. Should I wait to finish 300000 round?

Then, I try to run  ../../core/build/train_score -oscript train_obj.lua -omodel obj_model_init.net -sscript train_score.lua and  ../../core/build/train_ransac -oscript train_obj.lua -omodel obj_model_init.net -sscript train_score.lua -smodel score_model_init.net
Both of them stagnated like following.

For pre-train score regression CNN:

Successfully loaded sensor transformation:
[0.9998282978278875, 0.008186003849805841, 0.01662420535123559, -0.01393324413905143;
  -0.008090415588790156, 0.9999503986429169, -0.005809081637597863, 0.05228905046770047;
  -0.01667093393273896, 0.005673587475537798, 0.9998449331606215, 0.02712006871814571;
  0, 0, 0, 1]

Parsing config file: default.config
focal length: 525
x shift: 0
y shift: 0
raw data (rescale rgb): 1
secondary focal length: 585
raw x shift: 0
raw y shift: 0
image width: 640
image height: 480
ransac iterations: 256
ransac iterations (refinement): 8
ransac batch size: 100
ransac refinement gradient sub sampling: 0.01
ransac inlier threshold: 10
ransac inlier threshold: 100
random draw: 1
object script: train_obj.lua

Loading training set ...
Reading file names... 
Loading script: train_obj.lua
TORCH: Loading network from file: obj_model_init.net
TORCH: Set model to evaluation mode.
Loading script: train_score.lua
TORCH: Creating network.
TORCH: Set model to evaluation mode.
Choosing validation data.
Starting training round 1 of 80
TORCH: Set model to training mode.
TORCH: Doing a forward pass for 1600 patches.
CNN prediction took 0.692021s.

For end-to-end training using DSAC:

Successfully loaded sensor transformation:
[0.9998282978278875, 0.008186003849805841, 0.01662420535123559, -0.01393324413905143;
  -0.008090415588790156, 0.9999503986429169, -0.005809081637597863, 0.05228905046770047;
  -0.01667093393273896, 0.005673587475537798, 0.9998449331606215, 0.02712006871814571;
  0, 0, 0, 1]

Parsing config file: default.config
focal length: 525
x shift: 0
y shift: 0
raw data (rescale rgb): 1
secondary focal length: 585
raw x shift: 0
raw y shift: 0
image width: 640
image height: 480
ransac iterations: 256
ransac iterations (refinement): 8
ransac batch size: 100
ransac refinement gradient sub sampling: 0.01
ransac inlier threshold: 10
ransac inlier threshold: 100
random draw: 1
object script: train_obj.lua
object model: obj_model_init.net
score script: train_score.lua
score model: score_model_init.net

Loading training set ...
Reading file names... 
Loading script: train_score.lua
TORCH: Loading network from file: score_model_init.net
TORCH: Set model to evaluation mode.
Loading script: train_obj.lua
TORCH: Loading network from file: obj_model_init.net
TORCH: Set model to evaluation mode.
Round 0 of 10000.
TORCH: Set model to training mode.
TORCH: Set model to training mode.
Predicting object coordinates.
TORCH: Doing a forward pass for 1600 patches.
CNN prediction took 0.62037s.
Sampling 256 hypotheses.

I would appreciate if you could give me some advises about possible mistakes or anything that I might missed.

Jeff

**Comment by ebrach** [comment]
Hi Jeff,

well, you should wait for the 300000 updates for the best performance. If you stop too early it will not work, because the initialization is too bad. After how many update did you stop? If I remember correctly it took 1-2 days for us on a Tesla K80. I would assume with a recent graphic card (TitanX) it trains much faster.

I think your issues with train_score and train_ransac might stem from insufficient memory of your graphics card. If this is the case, you have several options:

1) Switch to a better graphics card ;)
2) Reduce the batch size in the train_obj.lua script, line 8: try values like 800 or 80
3) For train_ransac: Reduce the number of hypothesis (but this will probably reduce accuracy): append "-rI 128" (or smaller numbers) to your command line call

Hope that helps!
Eric
